compiler/src/main/java/com/squareup/anvil/compiler/codegen/incremental/GeneratedFileCache.kt

```diff
@@ -21,150 +22,169 @@ internal typealias FileId = Int
 
 internal class GeneratedFileCache private constructor(
   private val binaryFile: File,
-  private val projectDir: ProjectDir,
+  private val projectDir: BaseDir.ProjectDir,
+  private val buildDir: BaseDir.BuildDir,
 ) : Closeable {
 
   private val tables: Tables by lazy(NONE) { Tables.fromFile(binaryFile) }
 
-  /**
-   * Source files that don't have any source files of their own.
-   */
-  val rootSourceFiles: Set<RelativeFile>
+  /** Source files that don't have any source files of their own. */
+  val rootSourceFiles: Set<AbsoluteFile>
     get() = tables.getSourceFilesWithoutGeneratedContent()
-      .filter { source -> source != RelativeFile.ANY }
-      .toSet()
+      .mapNotNullToSet { it.absolute() }
```

Comment:
Good question.  It's happening implicitly elsewhere, but it also doesn't matter since `ANY_ABSOLUTE` actually gets added to this set downstream in case it didn't exist already.